cosmos/x/evm/plugins/historical/historical_data.go

```diff
@@ -106,6 +106,11 @@ func (p *plugin) GetBlockByNumber(number uint64) (*coretypes.Block, error) {
 		return nil, err
 	}
 
+	// no receipts/txs on genesis case, return early with just header
+	if number == 0 {
+		return coretypes.NewBlock(header, nil, nil, nil, trie.NewStackTrie(nil)), nil
```

Comment:
i don't like this optimization.

Reply:
unmarshaling receipts and txs with nil bytes (bc theyre not in the store) will cause this func to return an error. Have to avoid this by just passing in nil and nil into `NewBlock`